### ISSUE TITLE
checking if PCH take effect

### ISSUE THREAD

**Issue by filiatra** [sakra/cotire#57]
Hi, 

I incoporated cotire in my cmake project, I get the "gch" file created correctly.
However I see no reduction in compilation time (in fact there is some increase for creating the gch).

The verbose output shows the additional flags: -Winvalid-pch -include "7path-to/file_prefix.h" during compilation of the source files.

My feeling is that the PCH are created but not used. How can I check if the compiler used the PCH ?

I am using GCC 4.8.3,

thank you in advance
filiatra

**Comment by filiatra** [comment]
Update: I used the "-H" flag, and indeed I see (when compiling zlib with a miminal cmakelists):

[100%] Building C object CMakeFiles/zlib.dir/zutil.c.o
cd /usr/bin/cc  -DMY_ZCALLOC -H -O2 -g -DNDEBUG -fPIC -Winvalid-pch -include "zlib/cotire/zlib_C_prefix.h"  -o CMakeFiles/zlib.dir/zutil.c.o   -c zlib/zutil.c
x zlib/cotire/zlib_C_prefix.h.gch
. /zlib/zutil.h
...

The "x" means that the PCH is not used..
best
filiatra

**Comment by filiatra** [comment]
Update 2:
Before passing my target to cotire I have 
set_target_properties(zlib PROPERTIES POSITION_INDEPENDENT_CODE ON)

This puts -fPIC to my C flags, but cotire did not add this to the gch compilation line.

Is this on purpose, or a bug ?

thnx
filiatra

**Comment by sakra** [comment]
Looks like a bug. Please gist the output of a `cmake -DCotire_DEBUG=On` run.
It is also interesting that the `-Winvalid-pch` did not warn you about the incompatible precompiled header.

**Comment by filiatra** [comment]
here it is:
https://gist.github.com/filiatra/5802bf251083dcbbf5d2

**Comment by sakra** [comment]
Please try the new cotire version 1.7.1 released today. I have removed a lot of obsolete code for older versions of CMake including handling of PIC.

**Comment by filiatra** [comment]
Thank you! version 1.7.1 solvew this issue.

Another issue: I would like to use a common (automatically generated) PCH for many (eg. 10) targets.
Following the maual I set the target property COTIRE_CXX_PREFIX_HEADER_INIT.

This works, but what actually happens is that still new PCHs are created per target, all of them containing a single #include line, which includes the file I set at COTIRE_CXX_PREFIX_HEADER_INIT.

Then, all of the 10 PCHs are compiled, resulting in 10 times the same gch file, with 10 different names.

Probably it would be possible to really have a single gch for all targets ?
